### PR TITLE
internal/charmstore: implement blobstore upload GC

### PR DESCRIPTION
**Pull request by rogpeppe**
We start a worker that will sporadically delete uploads that
have passed their expiry date.

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2062/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2063/

File: internal/charmstore/blobstoregc.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+// collector worker.
+func newBlobstoreGC(pool *Pool) *blobstoreGC {
+	gc := &blobstoreGC{
+		pool: pool,
+	}
+	gc.tomb.Go(gc.run)
+	return gc
+}
+
+// Kill implements worker.Worker.Kill.
```
Comment (frankban) [inline internal/charmstore/blobstoregc.go:31]: Maybe indicate the full import path of worker?

File: internal/charmstore/blobstoregc.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+func (gc *blobstoreGC) Kill() {
+	gc.tomb.Kill(nil)
+}
+
+// Kill implements worker.Worker.Wait.
```
Comment (frankban) [inline internal/charmstore/blobstoregc.go:36]: Ditto, and also s/Kill/Wait/.

File: internal/charmstore/blobstoregc.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package charmstore
```
Comment (fabricematrat) [inline internal/charmstore/blobstoregc.go:4]: licence

**Review by frankban — APPROVED** [review]
Looks good with a couple of minors.

**Review by fabricematrat — APPROVED** [review]

File: internal/charmstore/blobstoregc.go (hunk 1 above)
Comment (rogpeppe) [reply internal/charmstore/blobstoregc.go:31]: I don't think it's worth it tbh.

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci-gce.jujugui.org:8080/job/charmstore-merge

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-cge.jujugui.org:8080//job/charmstore/2066/

**Pull request merged**